executor/index_lookup_hash_join.go

```diff
@@ -506,7 +506,7 @@ func (iw *indexHashJoinInnerWorker) getNewJoinResult(ctx context.Context) (*inde
 
 func (iw *indexHashJoinInnerWorker) buildHashTableForOuterResult(ctx context.Context, task *indexHashJoinTask, h hash.Hash64) {
 	buf, numChks := make([]byte, 1), task.outerResult.NumChunks()
-	task.lookupMap = newRowHashMap(task.outerResult.Len())
+	task.lookupMap = newUnsafeHashTable(task.outerResult.Len(), true)
```

Comment:
Do we really need to keep order here? IMO, Index Hash Join just cares about the order of the outer table.

Reply:
remove `keepOrder`